bots/pr/src/main/java/org/openjdk/skara/bots/pr/LimitedCensusInstance.java

```diff
@@ -55,27 +66,41 @@ private static Namespace namespace(Census census, String hostNamespace) {
         if (namespace == null) {
             throw new RuntimeException("Namespace not found in census: " + hostNamespace);
         }
-
         return namespace;
     }
 
-    private static Optional<JCheckConfiguration> configuration(HostedRepositoryPool hostedRepositoryPool, HostedRepository remoteRepo, String name, String ref) throws IOException {
-        return hostedRepositoryPool.lines(remoteRepo, Path.of(name), ref).map(JCheckConfiguration::parse);
+    private static Optional<JCheckConfiguration> configuration(HostedRepository remoteRepo, String name, String ref) {
+        Optional<List<String>> conf = Optional.empty();
+        try {
+            conf = Optional.of(Arrays.stream(remoteRepo.fileContents(name, ref).split("\n")).toList());
+        } catch (NoSuchElementException ignored) {
+            // NoSuchElementException will only work for tests
```

Comment:
I'm not sure I understand this part.

Reply:
Can we change `TestHostedRepository` to behave more like other remote repo implementations in this regard? It could throw UncheckedRestException with 404 if the file doesn't exist.

Reply:
Yes, it's a good idea!